pkg/agent/controller/egress/ip_assign_advertise_linux.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package egress
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/util"
+	"antrea.io/antrea/pkg/agent/util/arping"
+)
+
+type nodeEgressAssigner struct {
+	nodeConfig *config.NodeConfig
+}
+
+func NewNodeEgressIPAssigner(c *config.NodeConfig) *nodeEgressAssigner {
+	return &nodeEgressAssigner{
+		c,
+	}
+}
+
+func (a *nodeEgressAssigner) AssignOwnerNodeEgressIP(egressIp string) error {
+	return assignOwnerNodeEgressIP(egressIp, a.nodeConfig)
+}
+
+func (a *nodeEgressAssigner) UnAssignOwnerNodeEgressIP(egressIp string) error {
+	return unAssignOwnerNodeEgressIP(egressIp, a.nodeConfig)
+}
+
+func assignOwnerNodeEgressIP(egressIP string, nodeConfig *config.NodeConfig) error {
+	addr, link, err := getNodeLinkAddr(egressIP, nodeConfig.NodeIPAddr.IP)
+	if err != nil {
+		return err
+	}
+	if err := assignEgressIP(addr, link); err != nil {
+		return err
+	}
+	return nil
+}
+
+func unAssignOwnerNodeEgressIP(egressIP string, nodeConfig *config.NodeConfig) error {
+	addr, link, err := getNodeLinkAddr(egressIP, nodeConfig.NodeIPAddr.IP)
+	if err != nil {
+		return err
+	}
+	if err := unassignEgressIP(addr, link); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getNodeLinkAddr(egressIP string, nodeIPAddr net.IP) (*netlink.Addr, netlink.Link, error) {
+	egressSpecIP := net.ParseIP(egressIP)
+	localAddr, localIntf, err := util.GetIPNetDeviceFromIP(nodeIPAddr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get IPNetDevice from ip %v error: %+v", nodeIPAddr, err)
+	}
+	link, err := netlink.LinkByName(localIntf.Name)
+	if err != nil {
+		return nil, nil, fmt.Errorf("get netlink by name(%s) error: %+v", localIntf.Name, err)
+	}
+	addr := netlink.Addr{IPNet: &net.IPNet{IP: egressSpecIP, Mask: localAddr.Mask}}
```

Comment:
Does "/32" as mask work? Otherwise extra route will be added to the node if the egress IP is not in the same subnet of node IP. I think we should avoid impacting the node's other network configuration.

Reply:
done